src/main.rs

```diff
@@ -426,6 +442,17 @@ async fn main() -> anyhow::Result<()> {
             dispatch
                 .try_init()
                 .context("failed to init observability for host")?;
+
+            // Now that tracing is initialized, log the effective OTEL configuration
+            tracing::debug!(
+                ?otel_config,
+                traces_endpoint = %otel_config.traces_endpoint(),
+                metrics_endpoint = %otel_config.metrics_endpoint(),
+                logs_endpoint = %otel_config.logs_endpoint(),
+                protocol = ?otel_config.protocol,
+                "Effective OpenTelemetry configuration after merging CLI args, environment variables, and defaults"
```

Comment:
```suggestion
                traces_endpoint = %otel_config.traces_endpoint(),
                metrics_endpoint = %otel_config.metrics_endpoint(),
                logs_endpoint = %otel_config.logs_endpoint(),
                protocol = ?otel_config.protocol,
                "combined OpenTelemetry configuration (cli > env > defaults)"
```

Side note: this is a super vague question maybe, but is it common for people to have creds in their otel endpoints? Like if for some reason an endpoint was protected by HTTP basic auth or something.